simulation/state.go

```diff
@@ -166,10 +135,93 @@ func AppStateFn(
 		// we should replace the EvmDenom with BondDenom
 		evmState.Params.EvmDenom = stakingState.Params.BondDenom
 
+		observers := make([]string, 0)
+		// Get all the operator addresses of the validators.
+		// The observer set can be a subset of the validator set
+		for _, validator := range stakingState.Validators {
+			accAddress, err := observertypes.GetAccAddressFromOperatorAddress(validator.OperatorAddress)
+			if err != nil {
+				continue
+			}
+			observers = append(observers, accAddress.String())
+		}
+
+		// Shuffle the observers list
+		r.Shuffle(len(observers), func(i, j int) {
+			observers[i], observers[j] = observers[j], observers[i]
+		})
+
+		// Pick a random number of observers to add to the observer set
+		numObservers := r.Intn(11) + 5
+		if numObservers > len(observers) {
+			numObservers = len(observers)
+		}
+		observers = observers[:numObservers]
+
+		// update the observer genesis state
+		observerStateBz, ok := rawState[observertypes.ModuleName]
+		if !ok {
+			panic("observer genesis state is missing")
+		}
+		observerState := new(observertypes.GenesisState)
+		cdc.MustUnmarshalJSON(observerStateBz, observerState)
+		observerState.Observers.ObserverList = observers
+		observerState.CrosschainFlags.IsInboundEnabled = true
+		observerState.CrosschainFlags.IsOutboundEnabled = true
+
+		tss := sample.TSSRandom(r)
+		tss.OperatorAddressList = observers
+		observerState.Tss = &tss
+
+		// Pick a random account to be the admin of all policies
+		randomAccount := accs[r.Intn(len(accs))]
+		authorityStateBz, ok := rawState[authoritytypes.ModuleName]
+		if !ok {
+			panic("authority genesis state is missing")
+		}
+
+		// update the authority genesis state
+		authorityState := new(authoritytypes.GenesisState)
+		cdc.MustUnmarshalJSON(authorityStateBz, authorityState)
+		policies := authoritytypes.Policies{
+			Items: []*authoritytypes.Policy{
+				{
+					Address:    randomAccount.Address.String(),
+					PolicyType: authoritytypes.PolicyType_groupEmergency,
+				},
+				{
+					Address:    randomAccount.Address.String(),
+					PolicyType: authoritytypes.PolicyType_groupAdmin,
+				},
+				{
+					Address:    randomAccount.Address.String(),
+					PolicyType: authoritytypes.PolicyType_groupOperational,
+				},
+			},
+		}
+		authorityState.Policies = policies
+
+		//Update the fungible genesis state
+		fungibleStateBz, ok := rawState[fungibletypes.ModuleName]
+		if !ok {
+			panic("fungible genesis state is missing")
+		}
+		fungibleState := new(fungibletypes.GenesisState)
+		cdc.MustUnmarshalJSON(fungibleStateBz, fungibleState)
+		// TOODO generate ethereum address from r
```

Comment:
```suggestion
		// TODO generate ethereum address from r
```
What does it mean? We're using `sample.EthAddressRandom(r)` here?